app/controllers/api/sessions_controller.rb

```diff
@@ -0,0 +1,28 @@
+class Api::SessionsController < Api::BaseController
+  skip_before_action :require_login!, only: [:create], raise: false
+
+  def create
+    parameters = params[:user_login]
+    resource = User.find_for_database_authentication(email: parameters[:email])
+    resource ||= User.new
+
+    if resource.valid_password?(parameters[:password])
+      auth_token = resource.generate_auth_token
```

Comment:
why'd users need to perform login for API calls? wouldn't that easier to generate API token within a profile page?